lib/erb.rb

```diff
@@ -516,15 +516,15 @@ def scan
 
       class ExplicitScanner < Scanner # :nodoc:
         def scan
-          stag_reg = /(.*?)(^[ \t]*<%-|<%-|#{stags.join('|')}|\z)/m
+          stag_reg = /(.*?)(^[ \t]*<%-=?|<%-=?|#{stags.join('|')}|\z)/m
           etag_reg = /(.*?)(-%>|#{etags.join('|')}|\z)/m
           scanner = StringScanner.new(@src)
           while ! scanner.eos?
             scanner.scan(@stag ? etag_reg : stag_reg)
             yield(scanner[1])
 
             elem = scanner[2]
-            if /[ \t]*<%-/ =~ elem
+            if /[ \t]*<%-$/ =~ elem
```

Comment:
It seems like you're implementing `<%-` and `<%-=` inconsistently. `<%-` is trimming the space here, and `<%-=` is passing it to another place. Can you try applying the following patch:

```suggestion
            if /[ \t]*<%-=/ =~ elem
              yield('<%=')
            elsif /[ \t]*<%-/ =~ elem
```

and fix other places accordingly?

Reply:
@k0kubun I mentioned why I did such an inconsistent change:
> <%-= performs different outputs **based on the result of ruby expression**: it only trims the spaces when the result of ruby expression is nil or ''. So that its implementation is a bit different with the other stags.

So that it cannot simply trim content at that position. 
If I change it to your way, it won't support my scenario and I think it becomes useless.

Reply:
Ultimately, this was about https://github.com/ruby/erb/pull/31#discussion_r1032838440. Let's discuss that there instead.